src/HonzaBotner.Discord/Command/IChatCommand.cs

```diff
@@ -7,6 +8,8 @@ namespace HonzaBotner.Discord.Command
 {
     public interface IChatCommand
     {
-        Task Execute(DiscordClient client, DiscordMessage message, CancellationToken cancellationToken);
+        IEnumerable<string> RequiredPermissions { get; }
```

Comment:
This is not best change. The IEnumerable interface is not good for these things. You could write code which would be executed every time you access this property. I strongly suggest that you use ImmutableList.